storage/diemdb/src/diemdb_test.rs

```diff
@@ -481,6 +481,13 @@ fn verify_committed_transactions(
 
         // Fetch and verify transaction itself.
         let txn = txn_to_commit.transaction().as_signed_user_txn().unwrap();
+        let txn_with_proof = db
+            .get_transaction_by_hash(txn_to_commit.transaction().hash(), ledger_version, true)
+            .unwrap()
+            .unwrap();
+        txn_with_proof
```

Comment:
Probably `assert_eq!(txn_with_proof.transaction().hash(), txn_to_commit.transaction().hash())` as well, which is the user intended way to use it? @xli